include/swift/SIL/SILValue.h

```diff
@@ -566,150 +571,50 @@ inline bool ValueOwnershipKind::isCompatibleWith(SILValue other) const {
+            lifetimeConstraint == UseLifetimeConstraint::NonLifetimeEnding) &&
+           "ValueOwnershipKind::None can never have their lifetime ended");
   }
 
-  /// Return a map that is compatible with any and all ValueOwnershipKinds
-  /// except for \p kind.
-  static OperandOwnershipKindMap
-  compatibleWithAllExcept(ValueOwnershipKind kind) {
-    OperandOwnershipKindMap map;
-    unsigned index = 0;
-    unsigned end = unsigned(OwnershipKind::LastValueOwnershipKind) + 1;
-    for (; index != end; ++index) {
-      if (ValueOwnershipKind(index) == kind) {
-        continue;
-      }
-      map.add(ValueOwnershipKind(index),
-              UseLifetimeConstraint::NonLifetimeEnding);
-    }
-    return map;
+  OwnershipKind getPreferredKind() const {
+    return ownershipKind;
   }
 
-  /// Create a map that has compatibility constraints for each of the
-  /// ValueOwnershipKind, UseLifetimeConstraints in \p args.
-  static OperandOwnershipKindMap
-  compatibilityMap(std::initializer_list<
-                   std::pair<ValueOwnershipKind, UseLifetimeConstraint>>
-                       args) {
-    OperandOwnershipKindMap map;
-    for (auto &p : args) {
-      map.addCompatibilityConstraint(p.first, p.second);
-    }
-    return map;
+  bool isLifetimeEnding() const {
+    return lifetimeConstraint == UseLifetimeConstraint::LifetimeEnding;
   }
 
-  /// Return a map that states that an operand can take any ownership with each
-  /// ownership having a must be live constraint.
-  static OperandOwnershipKindMap allLive() {
-    OperandOwnershipKindMap map;
-    unsigned index = 0;
-    unsigned end = unsigned(OwnershipKind::LastValueOwnershipKind) + 1;
-    while (index != end) {
-      map.add(ValueOwnershipKind(index),
-              UseLifetimeConstraint::NonLifetimeEnding);
-      ++index;
-    }
-    return map;
+  UseLifetimeConstraint getLifetimeConstraint() const {
+    return lifetimeConstraint;
   }
 
-  /// Specify that the operand associated with this set can accept a value with
-  /// ValueOwnershipKind \p kind. The value provided by the operand will have a
-  /// new ownership enforced constraint defined by \p constraint.
-  void add(ValueOwnershipKind kind, UseLifetimeConstraint constraint) {
-    unsigned index = unsigned(kind);
-    unsigned kindOffset = index * 2;
-    unsigned constraintOffset = index * 2 + 1;
-
-    // If we have already put this kind into the map, we require the constraint
-    // offset to be the same, i.e. we only allow for a kind to be added twice if
-    // the constraint is idempotent. We assert otherwise.
-    assert((!data[kindOffset] || UseLifetimeConstraint(bool(
-                                     data[constraintOffset])) == constraint) &&
-           "Adding kind twice to the map with different constraints?!");
-    data[kindOffset] = true;
-    data[constraintOffset] = bool(constraint);
+  static OwnershipConstraint anyValueAcceptingConstraint() {
```

Comment:
I have a lot of difficulty parsing names like this and attempting to sort out their meaning. It accepts any value? Obviously not. "Constraint" is already in the type name, so why is it repeated?

Whereas if it were simply `any()` or even `unconstrained()` I would immediately know what that means.

I would also know what "acceptsAnyOwnership" means, so that would be ok. It's still unnecessarily verbose because constraints always imply acceptance and Ownership is already in the name.

Starting from a minimal recognizable name, before adding any more words, the question should be "how could any reasonable person be misled by the shorter name, and how would adding more words fix that problem without creating more confusion"?

Reply:
I think that any works well.